### PR TITLE
[Google Fonts] Add automated tests for loading fonts and calling font methods.

### PR DESCRIPTION
**Pull request by johnsonmh**
* Added tests for when http requests should and should not be made.
* Added tests for the scoring logic of _which_ font family is used given a `TextStyle`.
* Added a travis script to run tests, anazlyer, and formatter.

### CODE REVIEW

File: google_fonts/test/font_methods_test.dart [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import 'package:google_fonts/src/google_fonts_base.dart';
+import 'package:mockito/mockito.dart';
+import 'package:http/http.dart' as http;
+
+class MockHttpClient extends Mock implements http.Client {}
+
+main() {
+  setUp(() async {
+    httpClient = MockHttpClient();
+    when(httpClient.get(any)).thenAnswer((_) async {
```
Comment (clocksmith) [inline google_fonts/test/font_methods_test.dart]: Do these tests need a MockHttpClient?

File: google_fonts/test/font_methods_test.dart [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      fontWeight: FontWeight.w400,
+      fontStyle: FontStyle.italic,
+    );
+
+    final outputTextStyle = GoogleFonts.roboto(textStyle: inputTextStyle);
+
+    expect(outputTextStyle.fontFamily, equals('Roboto400italic'));
+  });
+
+  testWidgets('Text style no direct match picks closest font weight match',
```
Comment (clocksmith) [inline google_fonts/test/font_methods_test.dart]: Text style with?

File: google_fonts/test/font_methods_test.dart [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      fontWeight: FontWeight.w600,
+      fontStyle: FontStyle.normal,
+    );
+
+    final outputTextStyle = GoogleFonts.roboto(textStyle: inputTextStyle);
+
+    expect(outputTextStyle.fontFamily, equals('Roboto500normal'));
+  });
+
+  testWidgets('Italic text style no direct match picks closest match',
```
Comment (clocksmith) [inline google_fonts/test/font_methods_test.dart]: Italic text style with?

File: google_fonts/test/load_font_test.dart [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    // The following snippet pulled from
+    //  * https://flutter.dev/docs/cookbook/persistence/reading-writing-files#testing
+    final directory = await Directory.systemTemp.createTemp();
+    const MethodChannel('plugins.flutter.io/path_provider')
+        .setMockMethodCallHandler((methodCall) async {
+      if (methodCall.method == 'getApplicationDocumentsDirectory') {
+        return directory.path;
+      }
+      return null;
+    });
```
Comment (clocksmith) [inline google_fonts/test/load_font_test.dart]: nice

File: google_fonts/test/load_font_test.dart [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  });
+
+  tearDown(() {
+    clearCache();
+  });
+
+  testWidgets(' loadFont method calls http', (tester) async {
```
Comment (clocksmith) [inline google_fonts/test/load_font_test.dart]: nit: space

Comment (clocksmith) [reply google_fonts/test/load_font_test.dart]: nit: "loadFont (method) calls http get"

File: google_fonts/test/load_font_test.dart [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    final fakeFont = 'foo';
+    final fakeUrl = Uri.http('fonts.google.com', '/foo');
+
+    await loadFont(fakeFont, fakeUrl.toString());
+
+    verify(httpClient.get(fakeUrl)).called(1);
+  });
+
+  testWidgets('loadFont method does not make http request on subsequent calls',
```
Comment (clocksmith) [inline google_fonts/test/load_font_test.dart]: nit: http get request

File: google_fonts/test/load_font_test.dart [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    final fakeUrl = Uri.http('fonts.google.com', '/foo');
+
+    var directoryContents = await getApplicationDocumentsDirectory();
+    expect(directoryContents.listSync().isEmpty, isTrue);
+
+    await loadFont(fakeFont, fakeUrl.toString());
+    directoryContents = await getApplicationDocumentsDirectory();
+
+    expect(directoryContents.listSync().isNotEmpty, isTrue);
+    expect(
```
Comment (clocksmith) [inline google_fonts/test/load_font_test.dart]: nice

File: google_fonts/.travis.yml [hunk 8]
```diff
@@ -0,0 +1,34 @@
+os:
```
Comment (rami-a) [inline google_fonts/.travis.yml]: So there is already a travis.yml file in the root of this repo. I think you could probably just updated the travis script to include the google fonts project:

https://github.com/material-components/material-components-flutter-experimental/blob/master/travis_script.sh

File: google_fonts/test/font_methods_test.dart (hunk 1 above)
Comment (johnsonmh) [reply google_fonts/test/font_methods_test.dart]: No, it doesn't seem to need the mock client. 

I'm embarrassed to admit that I'm not sure how to be 100% certain that the tests aren't making real life http requests without the mock client though. 

I turned off my wifi and the tests still pass, so I guess that's good enough.

File: google_fonts/test/font_methods_test.dart (hunk 2 above)
Comment (johnsonmh) [reply google_fonts/test/font_methods_test.dart]: Done.

File: google_fonts/test/font_methods_test.dart (hunk 3 above)
Comment (johnsonmh) [reply google_fonts/test/font_methods_test.dart]: Done.

File: google_fonts/test/load_font_test.dart (hunk 5 above)
Comment (johnsonmh) [reply google_fonts/test/load_font_test.dart]: Done

File: google_fonts/test/load_font_test.dart (hunk 6 above)
Comment (johnsonmh) [reply google_fonts/test/load_font_test.dart]: Done.

File: google_fonts/test/font_methods_test.dart (hunk 2 above)
Comment (johnsonmh) [reply google_fonts/test/font_methods_test.dart]: Done

File: google_fonts/test/font_methods_test.dart (hunk 3 above)
Comment (johnsonmh) [reply google_fonts/test/font_methods_test.dart]: Done

File: google_fonts/test/load_font_test.dart (hunk 5 above)
Comment (johnsonmh) [reply google_fonts/test/load_font_test.dart]: Done

File: google_fonts/test/load_font_test.dart (hunk 6 above)
Comment (johnsonmh) [reply google_fonts/test/load_font_test.dart]: Done

File: google_fonts/test/load_font_test.dart (hunk 5 above)
Comment (johnsonmh) [reply google_fonts/test/load_font_test.dart]: Done

File: google_fonts/test/load_font_test.dart (hunk 6 above)
Comment (johnsonmh) [reply google_fonts/test/load_font_test.dart]: Done

Comment (johnsonmh) [reply google_fonts/test/load_font_test.dart]: Done

File: google_fonts/.travis.yml (hunk 8 above)
Comment (johnsonmh) [reply google_fonts/.travis.yml]: That worked, thanks!

**Pull request merged**